server/src/main/java/org/elasticsearch/action/admin/cluster/stats/AnalysisStats.java

```diff
@@ -133,7 +118,27 @@ public static AnalysisStats of(Metadata metadata, Runnable ensureNotCancelled) {
             Map<String, Settings> tokenFilterSettings = indexSettings.getGroups("index.analysis.filter");
             usedBuiltInTokenFilters.keySet().removeAll(tokenFilterSettings.keySet());
             aggregateAnalysisTypes(tokenFilterSettings.values(), usedTokenFilterTypes, indexTokenFilterTypes);
+            countMapping(mappingCounts, indexMetadata);
         }
+        for (Map.Entry<String, Integer> mappingAndCount : mappingCounts.entrySet()) {
+            Set<String> indexAnalyzers = new HashSet<>();
```

Comment:
I think we should also do `ensureNotCancelled.run()` here to ensure we can cancel here too.

Reply:
++ adding it back